freeglut/freeglut/src/fg_menu.c

```diff
@@ -780,14 +780,29 @@ void fghCalculateMenuBoxSize( void )
 /*
  * Creates a new menu object, adding it to the freeglut structure
  */
-int FGAPIENTRY glutCreateMenu( FGCBMenu callback )
+int FGAPIENTRY glutCreateMenuUcall( FGCBMenuUC callback, FGCBUserData userData )
 {
     /* The menu object creation code resides in fg_structure.c */
-    FREEGLUT_EXIT_IF_NOT_INITIALISED ( "glutCreateMenu" );
+    FREEGLUT_EXIT_IF_NOT_INITIALISED ( "glutCreateMenuUcall" );
     if (fgState.ActiveMenus)
         fgError("Menu manipulation not allowed while menus in use.");
 
-    return fgCreateMenu( callback )->ID;
+    return fgCreateMenu( callback, userData )->ID;
+}
+
+/* Standard glutCreateMenu */
+static void glutCreateMenuCallback( int menu, FGCBUserData userData )
```

Comment:
so if the original callback setter is called (e.g. glutCreateMenu in this case), you put an internal helper as the callback function and the user provided callback as the userData, and then internally handle calling the user callback without user data. Do i understand that correctly (and you use the same pattern in other places)?

- if this is an internal function, start it with fg or fgh (h for helper), not glut. all glutXXX() are exposed to users.
- can we generalize this helper, implement it once, and give it a better name showing its routing function (or what to call that)?